### DISCUSSION TITLE
Libary starts one folder higher than path

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#1737] [Bug Reports]
As identified in #1723 where I thought the Library issues stemmed from using Shared Data it now seems that 20240719 also exhits the same problem in that the Library Page starts to display albums at one folder too high in the path. My music library path is:

/mnt/MPD/NAS/NAS_Music

but the folder Album folder display seems to start at:

/mnt/MPD/NAS

The Library image in #1723 shows the problem quite clearly.(but in that example without the album covers die toe problem of Shared Data in #1723).

Chris

**Comment by ceejayemm** [comment]
This information is from an RPi3B+ running 20240719. The music library is mounted via NFS and shows correctly:

```
RAOffice:~  #  df -h
Filesystem                                 Size  Used Avail Use% Mounted on
/dev/root                                  6.9G  3.2G  3.5G  48% /
devtmpfs                                   419M     0  419M   0% /dev
tmpfs                                      453M   76K  453M   1% /dev/shm
tmpfs                                      182M  660K  181M   1% /run
tmpfs                                      453M  4.0K  453M   1% /tmp
/dev/mmcblk0p1                             197M   59M  139M  30% /boot
192.168.86.250:/export/music/iTunes/Music  4.3T  678G  3.6T  16% /mnt/MPD/NAS/NAS_Music
```

I have used this mount point for ages (years). A directory listing of this mount point shows thew first few folders as:

```
RAOffice:~  #  ls -la /mnt/MPD/NAS/NAS_Music
total 540
drwxrwsr-x+ 135 root   100  136 Jun  1 09:32  .
drwxrwxrwx    4 root  root 4096 Jul 11 08:49  ..
-rwxrwxr-x    1 root   100   38 Dec  1  2021  .mpdignore
drwxrwsr-x+   6 root   100    6 Dec  1  2021  ABBA
drwxrwsr-x+   3 root   100    3 Dec  1  2021 'Adrian Sutton'
drwxrwsr-x+   3 root   100    3 Dec  1  2021 'Agnetha F'$'\303\244''ltskog'
drwxrwsr-x+   3 root   100    3 Dec  1  2021 'Air Traffic Controller'
drwxrwsr-x+   3 root   100    3 Dec  1  2021 'Alexander Armstrong'
drwxrwsr-x+   3 root   100    3 Dec  1  2021 'Alfie Boe'
drwxrwsr-x+   3 root   100    3 Dec  1  2021 'Alison Moyet'
```

The rAudio Library page shows (truncated to left, so some of the folders above are not shown in this image):

![image](https://github.com/user-attachments/assets/9ef1b6db-d987-4278-a209-f30d1618421a)

where the top left default rAudio image refers to the folder '/mnt/MPD/NAS' and all the rest refer to folders under '/mnt/MPD/NAS/NAS_Music'

As noted previously (#1723) I thought this was a problem with the Shared Data features but I have recently built a brand new 'standard' installation (i20240615 / 20240719) where Shared Data is NOT used and it has this same problem.

Chris

**Reply by rern** [reply]
Hi Chris,

Please try the UPDATE.

**Reply by ceejayemm** [reply]
@rern 

The latest UPDATE branch seems to have solved this problem (and some of #1723 as well). BUT there seems to be a new issue when updating the Album thumbnails. The update got 90% of the way through the update and then halted with:

```
300/341 NAS/NAS_Music/Cliff Richard/Together with Cliff Richard
    !  No write permission: /mnt/MPD/NAS/NAS_Music/Cliff Richard/Together with Cliff Richard drwxrwsr-x (2775)
88% 00:01:08/00:01:17
301/341 NAS/NAS_Music/Toto/Toto IV
```
I have been using this share for a number of years and not had a problems with permissions before  (ALL the library update items had permission errors) but the Toto entry just froze.  Looking at journalctl -xe after this occurred showed:

```
-- The unit systemd-timedated.service has successfully entered the 'dead' state.
Aug 17 16:59:31 RAOffice nginx[446]: 2024/08/17 16:59:31 [error] 446#446: *401 FastCGI sent in stderr: "PHP message: 
PHP Warning:  Undefined variable $postinfo in /srv/http/settings/addonsprogress.php on line 33; PHP message: 
PHP Warning:  Undefined variable $installurl in /srv/http/settings/addonsprogress.php on line 114; 
PHP message: PHP Warning:  Undefined variable $installfile in /srv/http/settings/addonsprogress.php on line 117" 
while reading response header from upstream, client: 192.168.86.53, server: localhost, request: 
"POST /settings.php?p=addonsprogress HTTP/1.1", upstream: "fastcgi://unix:/run/php-fpm/php-fpm.sock:", host: "raoffice", 
referrer: "http://raoffice/"
Aug 17 16:59:31 RAOffice sudo[16366]:     http : PWD=/srv/http ; USER=root ; COMMAND=/srv/http/bash/albumthumbnail.sh  overwrite
Aug 17 16:59:31 RAOffice sudo[16366]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=33)
Aug 17 16:59:31 RAOffice nginx[446]: 2024/08/17 16:59:31 [error] 446#446: *401 FastCGI sent in stderr: "; PHP message: 
PHP Warning:  Undefined array key "pid " in /srv/http/settings/addonsprogress.php on line 151" while reading upstream, 
client: 192.168.86.53, server: localhost, request: "POST /settings.php?p=addonsprogress HTTP/1.1", upstream: 
"fastcgi://unix:/run/php-fpm/php-fpm.sock:", host: "raoffice", referrer: "http://raoffice/"
Aug 17 17:00:30 RAOffice kid3-cli[17434]: Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
                                          Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
                                          If this causes problems, reconfigure your locale. See the locale(1) manual
                                          for more information.
Aug 17 17:01:00 RAOffice CROND[17618]: (root) CMD (run-parts /etc/cron.hourly)
Aug 17 17:01:00 RAOffice CROND[17617]: (root) CMDEND (run-parts /etc/cron.hourly)
Aug 17 17:01:29 RAOffice sshd[17622]: Accepted password for root from 192.168.86.53 port 46472 ssh2
Aug 17 17:01:29 RAOffice sshd[17622]: pam_unix(sshd:session): session opened for user root(uid=0) by root(uid=0)
```

There seems to be a PHP error(s) followed by the kid3-cli error/warning which I guess occurs with the Toto album entry noted above.  I have had the Toto album for a while and never had any problems with it rebuilding the Library or the Artwork index.

If you need any further information please let me know.

Chris

**Comment by ceejayemm** [comment]
Fixed in 20240818 but a full rebuild of the Library seems necessary (understandably).

Thanks @rern
